src/updatedb.py

```diff
@@ -232,28 +232,28 @@ def uploadResults(data, branch, revision, date):
                         dir = (dir.split('|')[1]).strip()
                         if dir.endswith(filename):
                             dir = dir.split(filename)[0]
-                            failures.append(dir + '/' + filename)
-        # https://treeherder.mozilla.org/api/project/mozilla-central/jobs/1116367/
-        url = "https://treeherder.mozilla.org/api/project/%s/jobs/%s/" % (branch, _id)
-        data1 = fetch_json(url)
+                            failures.append(dir + filename)
+            # https://treeherder.mozilla.org/api/project/mozilla-central/jobs/1116367/
+            url = "https://treeherder.mozilla.org/api/project/%s/jobs/%s/" % (branch, _id)
+            data1 = fetch_json(url)
 
-        slave = data1['machine_name']
+            slave = data1['machine_name']
 
-        # Insert into MySQL Database
-        try:
-            testjob = Testjobs(str(slave), str(result), str(build_system_type),
-                               str(duration), str(platform), str(buildtype),
-                               str(testtype), str(bugid), str(branch), str(revision),
-                               str(date), str(failure_classification), str(failures))
-
-            session.add(testjob)
-            count += 1
-            session.commit()
-        except Exception as error:
-            session.rollback()
-            logging.warning(error)
-        finally:
-            session.close()
+            # Insert into MySQL Database
+            try:
+                testjob = Testjobs(str(slave), str(result), str(build_system_type),
+                                   str(duration), str(platform), str(buildtype),
+                                   str(testtype), str(bugid), str(branch), str(revision),
+                                   str(date), str(failure_classification), str(failures[0:2]))
```

Comment:
Because we may get some warning like:**WARNING:root:(_mysql_exceptions.DataError) (1406, "Data too long for column 'failures' at row 1")**  for a failures like:
"[u'toolkit/mozapps/extensions/test/browser/test-window//browser_bug596336.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_bug596336.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_bug596336.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_bug596336.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_bug596336.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_bug596336.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_bug596336.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_bug596336.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_experiments.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_install.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_install.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_install.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_install.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_install.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_install.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_list.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_list.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_list.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_list.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_list.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_searching.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_searching.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_searching.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_searching.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_searching.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_searching.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_searching.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_searching.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_searching.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_searching.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_sorting.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_sorting.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_sorting.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_sorting.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_sorting.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_uninstalling.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_uninstalling.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_uninstalling.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_uninstalling.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_uninstalling.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_uninstalling.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_uninstalling.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_uninstalling.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_uninstalling.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_uninstalling.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_webext_options.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_webext_options.js', u'toolkit/mozapps/extensions/test/browser/test-window//browser_webext_options.js']")